priv/repo/migrations/20200316103722_create_users_auth_tables.exs

```diff
@@ -0,0 +1,26 @@
+defmodule Demo.Repo.Migrations.CreateUsersAuthTables do
+  use Ecto.Migration
+
+  def change do
+    execute "CREATE EXTENSION IF NOT EXISTS citext", ""
+
+    create table(:users) do
+      add :email, :citext, null: false
+      add :hashed_password, :string, null: false
+      add :confirmed_at, :naive_datetime
```

Comment:
Would it be a good idea to break the `users` table up and move `:email` and `:confirmed_at` into a `users_contacts` table, and `:hashed_password` into a `users_auth_strategies` table? The `users` table would still remain for user-specific fields, ie. `username`, `name`, etc. but the tables could look something like the following:

```
create table(:user_contacts) do
  add :user_id, references(:users, on_delete: :delete_all), null: false
  add :context, :string, null: false
  add :value, :string, null: false
  add :confirmed_at, :naive_datetime
  timestamps()
end

create unique_index(:users_contacts, [:context, :value])

create table(:users_auth_strategies) do
  add :user_id, references(:users, on_delete: :delete_all), null: false
  add :context, :string, null: false
  add :hash, :string, null: false
  timestamps()
end

create unique_index(:users_auth_strategies, [:context, :user_id])
```

With the `user_contacts` table, it would be easier to support user authentication with an email, mobile phone number, or both. The `users_auth_strategies` table would hold the user's `:hashed_password` with a "local" `:context`, while also being able to support oauth-based contexts through github, twitter, google, facebook, etc.

Reply:
While this could be one direction, I decided to keep this to a minimum feature set and model it according to this minimum feature set. We could have made it more flexible, but then it would certainly become more complex. So it is not meant to answer all problems, but to be a starting point.

And of course, you can always expand on this on your own with your own example app or your own generators.

Reply:
Was hoping to increase applicability without adding too much complexity, but what you're saying makes sense. Thanks for the response.